app/controllers/api/v2/alerts_controller.rb

```diff
@@ -15,6 +15,18 @@ def index
     @alerts = @record.alerts
   end
 
+  def destroy
+    authorize! :update, @record
```

Comment:
Can you remove the :update permission check? The alert dismissal shows up on the show page, which means that someone without edit permissions but with remove alert permissions will be able to see the button. And this is a reasonable usecase to support: managers will likely want to dismiss alerts.